lisa/tools/hyperv.py

```diff
@@ -119,20 +120,21 @@ def create_vm(
                 force_run=True,
             )
 
-        # add disks
-        disk_info = powershell.run_cmdlet(
-            "(Get-Disk | Where-Object {$_.OperationalStatus -eq 'offline'}).Number",
-            force_run=True,
-        )
-        matched = re.findall(r"\d+", disk_info)
-        disk_numbers = [int(x) for x in matched]
-        for disk_number in disk_numbers:
-            self._run_hyperv_cmdlet(
-                "Add-VMHardDiskDrive",
-                f"-VMName {name} -DiskNumber {disk_number} -ControllerType 'SCSI'",
-                extra_args=extra_args,
+        # add disks if requested
+        if add_disks:
+            disk_info = powershell.run_cmdlet(
+                "(Get-Disk | Where-Object {$_.OperationalStatus -eq 'offline'}).Number",
```

Comment:
How is the number gotten? Is it possible to set to number = 0 to avoid add disks?

Reply:
This PowerShell command lists all offline disks and gets the count. `Number` is that count.

In our server, we have some extra disks lying around but we don't want to add them to the VMs. At least, not automatically.